src/main/java/io/opentracing/contrib/tracerresolver/TracerResolver.java

```diff
@@ -55,6 +55,14 @@
      * @return The resolved Tracer or {@code null} if none was resolved.
      */
     public static Tracer resolveTracer() {
+        try { // Take care NOT to import GlobalTracer as it is an optional dependency and may not be on the classpath.
+            if (io.opentracing.util.GlobalTracer.isRegistered()) {
+                return logResolved(io.opentracing.util.GlobalTracer.get());
+            }
+        } catch (NoClassDefFoundError globalTracerNotInClasspath) {
```

Comment:
@yurishkuro One of the intricate ways the classes are compiled; it will work as you mention if you add an 
```java
import io.opentracing.util.GlobalTracer;
```
at the top of the class. That will indeed cause `NoClassDefFoundError` upon loading.

Using the FQN without importing will throw it where it's used, allowing me to `catch` it properly (hence the comment about not importing).
I'll add an integrationtest (#17) to show and guarantee this behaviour.